### ISSUE TITLE
Video sometimes restarts from beginning when backgrounding

### ISSUE THREAD

**Issue by bobdanek** [TeamNewPipe/NewPipe#7916]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.22.0

### Steps to reproduce the bug

This is not the only way this is triggered, but it's how I've finally figured out how to reproduce the issue, though it does take a few attempts before it happens for me

1. Play YouTube video in the app, be somewhere far from the beginning of the video
2. Double-tap power button to bring up the camera (GrapheneOS' camera app)
3. Listen to the audio to confirm that it's started from the beginning - if it hasn't, goto step 1 and try again
4. Switch back to NewPipe and observe that your place in the video has been lost and that you're back at the beginning of the video

### Expected behavior

Background audio of a video I'm watching with NewPipe picks up where the video+audio left off when it goes into the background after I switch to another app, and when I return to NewPipe, the video+audio picks up where the background audio left off

### Actual behavior

Occasionally, when I switch to another app while watching a video in NewPipe, the audio restarts from the beginning of the video. And when I go back to the app, the video+audio picks up where that audio left off, at or close to the start of the video.

In other words, I lose my "place" in a video sometimes when switching apps.

### Screenshots/Screen recordings

n/a

### Logs

n/a

### Affected Android/Custom ROM version

GrapheneOS 2022021721

### Affected device model

Google Pixel 5

### Additional information

"Minimize on app switch" is set to "Minimize to background player"

**Comment by bobdanek** [comment]
My apologies, this is not the latest version. I'll update and try to reproduce, and update version number in this report if it still occurs.

**Comment by bobdanek** [comment]
issue also occurs in 0.22.0. I initially reported this for version 0.21.16, but edited the app version in the report so that it doesn't look like I lied when I checked the very first box about reproducing with the latest version. :P

**Comment by ImDeadGuy** [comment]
I've been encountering this for a few versions now, and think I finally found a fix!
My NewPipe's lost video position every time I switch from fullscreen to background or pop-up for a few good months.

v22.0 RC didn't do this, but updating the release from 21.16 to 22.0 still does it.

So I cleaned the app data, and behold, issue no longer happens.
I restored exported data + settings => issue comes back.
Cleaned data again => Issue solved.
Restored exported data but NOT settings => Issue stays fixed!

So I think this may have something to do with a legacy settings incompatibility, likely stored on the device for some builds now.

Hope this helps with both solving the issue locally and maybe debugging this issue at the source.
Thanks!

**Comment by mujibultanim** [comment]
video starts playing from the beginning, when I switch to pop-up player mode or Pop up player to full screen mode. This is kinda annoying 😕

**Comment by CyberaxAnnon** [comment]
Had this happen but thought it was just me. 
Been happening for months, hope this can get resolved.

**Comment by ImDeadGuy** [comment]
Thanks for confirming, Seedthis. This is now a 100% repro for me:
1) Play any video with the "Auto-enqueueing" toggle off (top-right in landscape mode)
2) Press the Popup button => Video popup starts at the current timestamp correctly.
3) Revert (maximize) to normal player
4) Toggle Auto-enqueueing to on
5) Press Popup button => Video popup starts at 0:0 timestamp

Toggling Auto-enqueue back to off does not fix it in the same session.
But it does if I force close and restart newpipe.
So it'd seem the auto-enqueue feature is the key here (and I did have that option on in the settings I did not restore 🤔 )

Bonus notes: the bug also happens when toggling Auto-enqueue Off->On->Off again without using the Popup feature when On.
Also, the bug happens both ways in the Popup switch (both Maximized -> Popup and Popup -> Maximized)

**Comment by mujibultanim** [comment]
It's not solved yet in new version, if I switch back to full screen from pop up video start playing from the beginning

**Comment by werjany** [comment]
> video starts playing from the beginning, when I switch to pop-up player mode or Pop up player to full screen mode. This is kinda annoying

This bug still happens all the time, version 0.22.1 Android 12 (MIUI 13).
